sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerSchemaTransformWriteProvider.java

```diff
@@ -0,0 +1,173 @@
+                                      Objects.requireNonNull(row))) {}))
+                  .apply(
+                      SpannerIO.write()
+                          .withDatabaseId(configuration.getDatabaseId())
+                          .withInstanceId(configuration.getInstanceId())
+                          .withFailureMode(SpannerIO.FailureMode.REPORT_FAILURES));
+          Schema failureSchema =
+              Schema.builder()
+                  .addStringField("operation")
+                  .addStringField("instanceId")
+                  .addStringField("databaseId")
+                  .addStringField("tableId")
+                  .addStringField("mutationData")
+                  .build();
+          result
+              .getFailedMutations()
+              .apply(
+                  FlatMapElements.into(TypeDescriptors.rows())
+                      .via(
+                          mtg ->
+                              Objects.requireNonNull(mtg).attached().stream()
+                                  .map(
+                                      mutation ->
+                                          Row.withSchema(failureSchema)
+                                              .addValue(mutation.getOperation().toString())
+                                              .addValue(configuration.getInstanceId())
+                                              .addValue(configuration.getDatabaseId())
+                                              .addValue(mutation.getTable())
+                                              // TODO(pabloem): Figure out how to represent
+                                              // mutation
+                                              //  contents in DLQ
+                                              .addValue(
+                                                  Iterators.toString(
+                                                      mutation.getValues().iterator()))
+                                              .build())
+                                  .collect(Collectors.toList())))
+              .setRowSchema(failureSchema);
+          return PCollectionRowTuple.empty(input.getPipeline());
+        }
+      };
+    }
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized String identifier() {
+    return "beam:schematransform:org.apache.beam:spanner_write:v1";
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      inputCollectionNames() {
+    return Collections.singletonList("input");
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      outputCollectionNames() {
+    return Collections.emptyList();
+  }
+
+  @AutoValue
+  @DefaultSchema(AutoValueSchema.class)
+  public abstract static class SpannerSchemaTransformWriteConfiguration implements Serializable {
```

Comment:
Any validation checks needed here?

Reply:
not for now. For now we will allow ourselves to fail at runtime, but we may add validations later.